internal/builders/bazel/rebuilder.sh

```diff
@@ -0,0 +1,490 @@
+verbose=0
+
+# Boolean to trigger cleanup upon completion or failure.
+cleanup=0
+
+# Outputs the usage of the Rebuilder script for the two modes:
+# 1) Verify and Rebuild
+# 2) Rebuild only
+function usage() {
+  if [[ $verify ]]
+  then
+    # Disabled to stop triggering warnings about color env vars.
+    # shellcheck disable=SC2059
+    printf "${RED}[ERROR] ${LIGHT_RED}Wrong usage. Usage to verify AND rebuild artifact:${RESET}\n"
+    printf "${CYAN}Usage: %s ${YELLOW}--artifact_path${RESET} <path> ${YELLOW}--prov_path${RESET} <path> ${YELLOW}--source_uri${RESET} <uri> ${YELLOW}--builder_id${RESET} <id> ${MAGENTA}[--docker_image]${RESET} <image> ${MAGENTA}[--verify]${RESET}\n" "$0"
+    printf "${RED}[ERROR] ${LIGHT_RED}Wrong usage. Usage to ONLY rebuild the artifact:${RESET}\n"
+    printf "${CYAN}Usage: %s ${YELLOW}--artifact_path${RESET} <path> ${YELLOW}--prov_path${RESET} <path> ${MAGENTA}[--docker_image]${RESET} <image>\n" "$0"
+  else
+    # shellcheck disable=SC2059
+    printf "${RED}[ERROR] ${LIGHT_RED}Wrong usage. Usage to ONLY rebuild the artifact:${RESET}\n"
+    printf "${CYAN}Usage: %s ${YELLOW}--artifact_path${RESET} <path> ${YELLOW}--prov_path${RESET} <path> ${MAGENTA}[--docker_image]${RESET} <image>\n" "$0"
+    printf "${RED}[ERROR] ${LIGHT_RED}Wrong usage. Usage to verify AND rebuild artifact:${RESET}\n"
+    printf "${CYAN}Usage: %s ${YELLOW}--artifact_path${RESET} <path> ${YELLOW}--prov_path${RESET} <path> ${YELLOW}--source_uri${RESET} <uri> ${YELLOW}--builder_id${RESET} <id> ${MAGENTA}[--docker_image]${RESET} <image> ${MAGENTA}[--verify]${RESET}\n" "$0"
+  fi
+}
+
+# Processes an argument for the script. Returns 0 if the given argument
+# was recognized as an argument for this script, and 1 if it was not.
+function process_argument() {
+  case "$1" in
+    --artifact_path=*) artifact_path="${1#--artifact_path=}" ;;
+    --prov_path=*) prov_path="${1#--prov_path=}" ;;
+    --source_uri=*) source_uri="${1#--source_uri=}" ;;
+    --builder_id=*) builder_id="${1#--builder_id=}" ;;
+    --docker_image=*) docker_image="${1#--docker_image=}" ;;
+    --verify) verify=1 ;;
+    --verbose) verbose=1 ;;
+    --cleanup) cleanup=1 ;;
+
+    *)
+      return 1 ;;
+  esac
+  return 0
+}
+
+# This function is will clean up built directories off after error.
+function cleanup() {
+  # If the cleanup flag is specified, remove created directories.
+  if [[ $cleanup -eq 1 ]]
+  then
+    type_writer "🧹---> Cleaning up $rebuilt_artifacts_dir..."
+    rm -rf $rebuilt_artifacts_dir
+
+    type_writer "🧹---> Cleaning up $repo_name..."
+    sudo rm -rf "$repo_name"
+
+    type_writer "🧹---> Cleaning up slsa-verifier..."
+    sudo rm -rf slsa-verifier
+  fi
+}
+
+# Parse arguments sequentially to check for unrecognized arguments
+for ARG in "$@"; do
+  returnValue=$?
+  process_argument "$ARG"
+  if [[ ! ($returnValue) ]]
+  then
+    my_arg="$ARG"
+    printf "${RED}[ERROR] ${LIGHT_RED}%s is unrecognized${RESET}\n" "$my_arg"
```

Comment:
if we want to use colors, we need to create a print_err function which abstracts away that functionality. Otherwise it's really hard to read